demisto_sdk/commands/pre_commit/hooks/docker.py

```diff
@@ -222,39 +262,39 @@ def prepare_hook(
                 for file in copy_files:
                     source: Path = CONTENT_PATH / file
                     target = obj.path.parent / Path(file).name
-                    if source != target and source.exists():
+                    if source != target and source.exists() and not target.exists():
                         shutil.copy(
                             CONTENT_PATH / file, obj.path.parent / Path(file).name
                         )
-        split_by_obj = self._get_property("split_by_object", False)
+        run_isolated = self._get_property("run_isolated", False)
         config_arg = self._get_config_file_arg()
         start_time = time.time()
         logger.debug(f"{len(tag_to_files_objs)} images were collected from files")
         logger.debug(f'collected images: {" ".join(tag_to_files_objs.keys())}')
-        for image, files_with_objects in sorted(
-            tag_to_files_objs.items(), key=lambda item: item[0]
-        ):
-            object_to_files = _split_by_objects(
-                files_with_objects, config_arg, split_by_obj
-            )
-            image_is_powershell = any(
-                obj.is_powershell for _, obj in files_with_objects
-            )
-
-            dev_image = devtest_image(image, image_is_powershell, self.context.dry_run)
-            hooks = self.get_new_hooks(
-                dev_image,
-                image,
-                object_to_files,
-                config_arg,
-            )
+        with ThreadPoolExecutor(max_workers=cpu_count()) as executor:
```

Comment:
arent we not pulling here? Whats the point of the threadpool?

Reply:
we make requests to dockerhub api to check if the image is availible.